circuits/cpp/barretenberg/cpp/src/barretenberg/benchmark/relations_bench/barycentric.bench.cpp

```diff
@@ -0,0 +1,29 @@
+#include "barretenberg/ecc/curves/bn254/fr.hpp"
+#include "barretenberg/polynomials/barycentric.hpp"
+#include <benchmark/benchmark.h>
+
+using namespace benchmark;
+
+namespace {
+auto& engine = numeric::random::get_debug_engine();
+}
+
+using FF = barretenberg::fr;
+using barretenberg::Univariate;
+using barretenberg::BarycentricData;
+
+
+namespace proof_system::benchmark {
+
+void extend_2_to_6(State& state) noexcept
+{
+    auto univariate = Univariate<FF, 2>::get_random();
+    BarycentricData<FF, 2, 6> barycentric_2_to_6;
+    for (auto _ : state)
+    {
+        DoNotOptimize(barycentric_2_to_6.extend(univariate));
```

Comment:
From https://github.com/google/benchmark/blob/main/docs/user_guide.md#preventing-optimization

> DoNotOptimize(<expr>) forces the result of <expr> to be stored in either memory or a register. 

Since the result of `extend` is not used, the compiler would most likely optimize away the function call, meaning at the benchmark would measure the cost of doing nothing.